numpy/testing/tests/test_utils.py

```diff
@@ -468,47 +468,70 @@ def test_report_fail_percentage(self):
         self.assertTrue("mismatch 25.0%" in msg)
 
 class TestArrayAlmostEqualNulp(unittest.TestCase):
-    @dec.knownfailureif(True, "Github issue #347")
-    def test_simple(self):
-        np.random.seed(12345)
-        for i in range(100):
-            dev = np.random.randn(10)
-            x = np.ones(10)
-            y = x + dev * np.finfo(np.float64).eps
-            assert_array_almost_equal_nulp(x, y, nulp=2 * np.max(dev))
-
-    def test_simple2(self):
-        x = np.random.randn(10)
-        y = 2 * x
-        def failure():
-            return assert_array_almost_equal_nulp(x, y,
-                                                  nulp=1000)
-        self.assertRaises(AssertionError, failure)
-
-    def test_big_float32(self):
-        x = (1e10 * np.random.randn(10)).astype(np.float32)
-        y = x + 1
-        assert_array_almost_equal_nulp(x, y, nulp=1000)
-
-    def test_big_float64(self):
-        x = 1e10 * np.random.randn(10)
-        y = x + 1
-        def failure():
-            assert_array_almost_equal_nulp(x, y, nulp=1000)
-        self.assertRaises(AssertionError, failure)
 
-    def test_complex(self):
-        x = np.random.randn(10) + 1j * np.random.randn(10)
-        y = x + 1
-        def failure():
-            assert_array_almost_equal_nulp(x, y, nulp=1000)
-        self.assertRaises(AssertionError, failure)
-
-    def test_complex2(self):
-        x = np.random.randn(10)
-        y = np.array(x, np.complex) + 1e-16 * np.random.randn(10)
+    def test_float64_pass(self):
+        nulp = 5
+        x = np.linspace(-20, 20, 50, dtype=np.float64)
+        x = 10**x
+        x = np.r_[-x, x]
+        eps = np.finfo(x.dtype).eps
+        y = x + x*eps*nulp/2.
```

Comment:
Looks like epsneg is just half of eps. It probably makes sense to use that for the negative case. I wouldn't add new test functions, just add new tests in the existing functions, the traceback on failure will indicate the relevant line. Note that complex comes in both float32/float64 forms also.